source/elements/oneMKL/source/domains/dft/descriptor.rst

```diff
@@ -417,6 +422,84 @@ type ``oneapi::mkl::dft::domain``, ``oneapi::mkl::dft::precision``,
 
 **Descriptor class member table:** :ref:`onemkl_dft_descriptor_member_table`
 
+.. _onemkl_dft_descriptor_set_workspace:
+
+set_workspace
++++++++++++++
+
+Sets the workspace for when ``config_param::WORKSPACE_PLACEMENT`` is set to ``config_value::WORKSPACE_EXTERNAL``.
+
+.. rubric:: Description
+
+This function sets the workspace to use when computing DFTs for when an
+external workspace is set. 
+This function may only be called after the descriptor has been committed.
+The size of the provided workspace must be equal to or larger than the required 
+workspace size obtained by calling ``descriptor<prec, dom>::get_value(config_param::WORKSPACE_EXTERNAL_BYTES, &workspaceBytes)``.
+
+A descriptor where ``config_value::WORKSPACE_EXTERNAL`` is specified for 
+``config_param::WORKSPACE_PLACEMENT`` is not a valid descriptor for compute 
+calls until this function has been successfully called.
+
+The type of workspace must match the compute calls for which it is used.
+That is, if the workspace is provided as a ``sycl::buffer``, the compute
+calls must also use ``sycl::buffer`` for their arguments. Likewise, a USM
+allocated workspace must only be used with USM compute calls.
+Failing to do this will result in an invalid descriptor for compute calls.
```

Comment:
Was there any specific reason for this limitation?

Reply:
Yes. The oneMKL (MKL-GPU) backend has this limitation.

A second reason is that the oneMKL interface treats the `sycl::buffer` and USM somewhat orthogonally. The buffer interface does not accept or return `sycl::event`s for instance. Mixing USM and buffers with it would be an odd choice to my mind.

Reply:
Right oneMKL does not support this. But I was trying to check that technically there is no limitation. I believe initially we wanted to support mixing in closed source but did not due to some eSIMD issues.

Reply:
Are you happy for this comment to be resolved?

Reply:
Yes, go ahead. Thank you